Framework/PythonInterface/mantid/plots/axesfunctions.py

```diff
@@ -630,27 +631,32 @@ def imshow(axes, workspace, *args, **kwargs):
         (aligned, kwargs) = check_resample_to_regular_grid(workspace, **kwargs)
         (normalize_by_bin_width, kwargs) = get_normalize_by_bin_width(workspace, axes, **kwargs)
         (distribution, kwargs) = get_distribution(workspace, **kwargs)
+        _setLabels2D(axes,
+                     workspace,
+                     transpose=transpose,
+                     normalize_by_bin_width=normalize_by_bin_width)
+        # sample data for large workspaces to reduce the amount of time it takes to plot
+        if workspace.getNumberHistograms() > 5000:
```

Comment:
May I ask if it is worth doing the sampling approach for all non MD workspaces? It would be a simpler thing to explain to users that all MatrixWorkspace types are resampled rather than say there is a number that over this it's done one way and under another.

Either way I think it would be good to move all of the code inside `samplingimage` and if we decide to have a threshold then it can be inside that module and we don't need to duplicate the code for pulling out the data etc.